src/record/record.js

```diff
@@ -237,9 +240,10 @@ Record.prototype.discard = function() {
 	this.whenReady( function() {
 		this.usages--;
 		if( this.usages <= 0 ) {
-				this.emit( 'destroyPending' );
-				this._discardTimeout = setTimeout( this._onTimeout.bind( this, C.EVENT.ACK_TIMEOUT ), this._options.subscriptionTimeout );
-				this._connection.sendMsg( C.TOPIC.RECORD, C.ACTIONS.UNSUBSCRIBE, [ this.name ] );
+			this.isReady = false;
+			this._discardTimeout = setTimeout( this._onTimeout.bind( this, C.EVENT.ACK_TIMEOUT ), this._options.subscriptionTimeout );
+			this._connection.sendMsg( C.TOPIC.RECORD, C.ACTIONS.UNSUBSCRIBE, [ this.name ] );
+	  	this._resubscribeNotifier.destroy();
```

Comment:
Not part of the original implementation. But I believe the resubscribeNotifier should be destroyed as soon as the `UNSUBSCRIBE`has been sent.